src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,14 +27,25 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        String[] components = trimmedAddress.split(", ");
+
+        Block blk = new Block (components[0]);
+        Street str = new Street (components[1]);
+        Unit uni = new Unit (components[2]);
+        PostCode poscode = new PostCode (components[3]);
```

Comment:
The attribute names here do not follow the naming convention. Always try to use understandable names instead of abbreviations so that other developers can understand.